### ISSUE TITLE
Field grid 

### ISSUE THREAD

**Issue by GJBoth** [TuragaLab/chromatix#8]
I'm not entirely convinced of the way we currently implement the different grids in the field; each grid type has his own function and is a property. That doens't make it very clear which grids we support and all the different functions don't help either. Looking for example at [numpy.norm](https://numpy.org/doc/stable/reference/generated/numpy.linalg.norm.html), they have a keyword 'ord' which sets which norm they're calculating. 

I'm proposing to combine all the functions into a single function, looking something like this:

```
grid(self, norm=None):
   if norm is None:
      return grid
   elif norm == "l2":
       return l2_norm(grid)
     ...
```

etc. This would solve all of the issues above. WDYT @diptodip ?

**Comment by diptodip** [comment]
I don't really see this as a problem. I'm not that opposed to the change though. That said, I would much rather type:
```python
field.l2_sq_grid
```
than:
```python
field.grid(norm='l2')
```
Maybe the first one could be renamed to `grid_l2_sq` for better completions or something.

Also, I'd prefer if either everything is a property or everything is a function, because then I don't have to think about whether or not to include parentheses (when I don't have completions).

**Comment by GJBoth** [comment]
Closed by #81
